### PR TITLE
CMR-7880 Adding FreeAndOpenData translations to ISO MENDS/SMAP also c…

### PR DESCRIPTION
**Pull request by eereiter**
…hanging AcccessConstraints for NSIDC.

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj [hunk 1]
```diff
@@ -93,23 +101,44 @@
                            ;; entire matching string is returned and if there is a group in the regular
                            ;; expression, the first group of the matching string is returned.
                            (if (string? match) match (second match))))]
-      (str/join matches))))
+      (string/join matches))))
+
+(defn parse-ac
+  "Parse the passed in ISO XML access constraints description and value using the
+   passed in xpath and regular expression in this function."
```
Comment (jaybarra) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj:108]: whitespace nitpick, style guide shows multi-line docstrings should start under the quote

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj [hunk 2]
```diff
@@ -93,23 +101,44 @@
...
+                           (parse-ac doc constraints-xpath other-constraints-xpath new-access-desc))
                        su/ACCESSCONSTRAINTS_DESCRIPTION_MAX
                        sanitize?)
          :Value (when value
-                 (Double/parseDouble value))}]
+                  (try (Double/parseDouble (string/trim value))
+                    (catch Exception e
+                      (warn (str "Exception thrown while trying to parse a non parsable ISO access "
+                                 "constraint value number. The document is " doc))
+                      nil)))}]
```
Comment (jaybarra) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj:142]: `nil` shouldn't be necessary here, warn already returns nil.

**Review by jaybarra — APPROVED** [review]

File: umm-spec-lib/test/cmr/umm_spec/test/xml_to_umm_mappings/iso_shared/use_constraints.clj [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+                             <gco:CharacterString>Access Constraints Value: 20</gco:CharacterString>
+                           </gmd:otherConstraints>
+                         </gmd:MD_LegalConstraints>
+                       </gmd:resourceConstraints>
+                       <gmd:aggregationInfo>")
+      smap-xpath
+      false
+      {:Value 15.0 :Description nil}
+
+      "Test Access Constraints Value constraints using smap for Restriction Flag.
```
Comment (ygliuvt) [inline umm-spec-lib/test/cmr/umm_spec/test/xml_to_umm_mappings/iso_shared/use_constraints.clj:186]: For completeness, we should have a test where the AC value is not a double.

**Review by ygliuvt — APPROVED** [review]

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj (hunk 1 above)
Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj:108]: I have update the docstring.

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj (hunk 2 above)
Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj:142]: I have removed the nil.

File: umm-spec-lib/test/cmr/umm_spec/test/xml_to_umm_mappings/iso_shared/use_constraints.clj (hunk 3 above)
Comment (eereiter) [reply umm-spec-lib/test/cmr/umm_spec/test/xml_to_umm_mappings/iso_shared/use_constraints.clj:186]: I do, the next test case is where the value is "abc" which is not a double. All of the other test cases, the value is actually an integer, but the code translates the number to a double.

**Pull request merged**